test/unit/puppetlabs/services/jruby/jruby_puppet_testutils.clj

```diff
@@ -181,3 +189,114 @@ create-mock-pool-instance :- JRubyInstance
     (while (< (count (jruby-core/registered-instances pool-context))
               num-jrubies)
       (Thread/sleep 100))))
+
+(schema/defn ^:always-validate
+  mock-puppet-config-settings :- {schema/Str schema/Any}
+  "Return a map of settings that mock the settings that core Ruby Puppet
+  would return via a call to JRubyPuppet.getSetting()."
+  [jruby-puppet-config :- {:master-conf-dir schema/Str
+                           :master-code-dir schema/Str
+                           schema/Keyword schema/Any}]
+  (let [certname "localhost"
+        confdir (:master-conf-dir jruby-puppet-config)
+        ssldir (str confdir "/ssl")
+        certdir (str ssldir "/certs")
+        cadir (str ssldir "/ca")
+        private-key-dir (str ssldir "/private_keys")]
+    {"allow_duplicate_certs" false
+     "autosign" true
+     "keylength" 2048
+     "cacert" (str cadir "/ca_crt.pem")
+     "ca_name" (str "Puppet CA: " certname)
+     "cacrl" (str cadir "/ca_crl.pem")
+     "cakey" (str cadir "/ca_key.pem")
+     "capub" (str cadir "/ca_pub.pem")
+     "ca_ttl" 157680000
+     "certdir" certdir
+     "certname" certname
+     "cert_inventory" (str cadir "/inventory.txt")
+     "codedir" (:master-code-dir jruby-puppet-config)
+     "csr_attributes" (str confdir "/csr_attributes.yaml")
+     "csrdir" (str cadir "/requests")
+     "data_binding_terminus" "hiera"
+     "dns_alt_names" ""
+     "hostcert" (str certdir "/" certname ".pem")
+     "hostcrl" (str ssldir "/crl.pem")
+     "hostprivkey" (str private-key-dir "/" certname ".pem")
+     "hostpubkey" (str ssldir "/public_keys/" certname ".pem")
+     "localcacert" (str certdir "/ca.pem")
+     "manage_internal_file_permissions" true
+     "privatekeydir" private-key-dir
+     "requestdir" (str ssldir "/certificate_requests")
+     "serial" (str cadir "/serial")
+     "signeddir" (str cadir "/signed")
+     "ssl_client_header" "HTTP_X_CLIENT_DN"
+     "ssl_client_verify_header" "HTTP_X_CLIENT_VERIFY"
+     "trusted_oid_mapping_file" (str confdir
+                                     "/custom_trusted_oid_mapping.yaml")}))
```

Comment:
Hmmm, this is a bummer.  It's probably better than any alternatives but it seems like it might be a bit of a bear to maintain.  I can definitely imagine people from feature teams coming in and adding new settings to the whitelist of stuff we pull from puppet, and not knowing that they need to update this - not sure if any tests would fail unless they happened to be using mock jrubies *and* exercising their new things in the same test?  Which seems like it wouldn't always be the case?

Do you think it might be worth having a memoized helper function that just loads up a scriptingcontainer and scrapes this stuff once?  Or loads up the puppet-server-config-service or whatever it's called, grabs the config from that, and memoizes it?  I know that'd probably add 10 seconds or something back to the overall test runtime but maybe that's worth it in exchange for the maintenance overhead of this chunk?

Not sure, just a thought.